dashboard/test/ui/features/step_definitions/blockly.rb

```diff
@@ -246,3 +246,25 @@ def current_block_xml
     return __TestInterface.getBlockXML();
   JS
 end
+
+def clear_main_block_space
+  # Do our async wait on the JavaScript side instead of polling over the wire.
+  # See https://seleniumhq.github.io/selenium/docs/api/java/org/openqa/selenium/JavascriptExecutor.html
+  # and https://www.rubydoc.info/gems/selenium-webdriver/3.8.0/Selenium/WebDriver/Driver#execute_async_script-instance_method
+  result = @browser.execute_async_script <<-JS
+    var callback = arguments[arguments.length - 1];
+    var waitStart = Date.now();
+    var timeoutMs = 5000;
+    (function checkForAndClearMainBlockSpace() {
+      if (Blockly.mainBlockSpace) {
+        Blockly.mainBlockSpace.clear();
+        callback();
+      } else if (Date.now() - waitStart > timeoutMs){
+        callback('Timeout reached: Unable to clear Blockly.mainBlockSpace');
+      } else {
+        setTimeout(checkForAndClearMainBlockSpace(), 100);
+      }
+    }())
+  JS
+  expect(result).to be_nil
```

Comment:
Note, although there's a Selenium script timeout, it gives a spew of confusing output when you hit it.  I'm implementing my own client-side timeout here (5 seconds seemed pretty good in this case where we weren't waiting at all before) that will result in a nice rspec assertion failures if it's reached.